fix(docs): Update AWS IAM Policy docs

toredash opened the pull request:
Missing ec2:DescribeTags after https://github.com/cilium/cilium/commit/e66ed7fee9513b7ac23b3eb8327679a8121dd6c2 got merged. This commit introduced a helper function which would require the AWS IAM Policy `ec2:DescribeTags` to be added to the cilium operator.

I'm not sure the full implications of not having this policy present, as it seems to be only used in certain cases. Either way, including this policy seems fair and I don't see any security implications of it. 

Please ensure your pull request adheres to the following guidelines:

- [X] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [ ] All code is covered by unit and/or runtime tests where feasible.
- [ ] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [ ] If your commit description contains a `Fixes: <commit-id>` tag, then
      please add the commit author[s] as reviewer[s] to this issue.
- [ ] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [ ] Provide a title or release-note blurb suitable for the release notes.
- [ ] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)
- [ ] Thanks for contributing!

<!-- Description of change -->

Commit https://github.com/cilium/cilium/commit/e66ed7fee9513b7ac23b3eb8327679a8121dd6c2 introduced a helper function for looking up EKS cluster name in AWS. This requires the IAM Policy `ec2:DescribeTags`, which is not documented.

This PR updates the documentation for required IAM Policy rights needed for Cilium to work in EKS.

```release-note
Update the documentation for required IAM policy rights needed for Cilium to work in EKS.
```

gandro commented on Documentation/network/concepts/ipam/eni.rst:
```diff
@@ -479,6 +479,7 @@ perform ENI creation and IP allocation:
  * ``ModifyNetworkInterfaceAttribute``
  * ``AssignPrivateIpAddresses``
  * ``CreateTags``
+ * ``DescribeTags``
```
Based [on the message](https://github.com/cilium/cilium/blob/8dabf8cea180102511e32243b3e62baa6d7dfdb7/pkg/ipam/allocator/aws/aws.go#L64-L66) that is shown whens describing the tags, there is a way to instead just use custom tags. Let's document that:

```suggestion
If ENI GC is enabled (which is the default), and ``--cluster-name`` and ``--eni-gc-tags`` are not set to custom values:

 * ``DescribeTags``
 
```

zacharysarah replied:
+1, with one comma nit
```suggestion
If ENI GC is enabled (which is the default) and ``--cluster-name`` and ``--eni-gc-tags`` are not set to custom values:

 * ``DescribeTags``
```

gandro reviewed (commented):
Nice find! Thanks a lot. I think we can document that the permission can be worked around, since it's not strictly required if the user manually specifies ENI GC tags.

christarazi reviewed (approved):
LGTM pending Sebastian's suggestion

zacharysarah reviewed (approved):
One non-blocking nit, otherwise LGTM from a docs perspective

gandro reviewed (approved):
Awesome, looks good, thanks a lot!

gandro commented on Documentation/network/concepts/ipam/eni.rst:
```diff
@@ -479,6 +479,9 @@ perform ENI creation and IP allocation:
  * ``ModifyNetworkInterfaceAttribute``
  * ``AssignPrivateIpAddresses``
  * ``CreateTags``
+If ENI GC is enabled (which is the default) and ``--cluster-name`` and ``--eni-gc-tags`` are not set to custom values:
```
It seems CI caught a formatting issue here (needs a new line between the list and the new paragraph)

`12:25:30 PM: /opt/build/repo/Documentation/network/concepts/ipam/eni.rst:482: WARNING: Block quote ends without a blank line; unexpected unindent.`

```suggestion

If ENI GC is enabled (which is the default) and ``--cluster-name`` and ``--eni-gc-tags`` are not set to custom values:
```

toredash replied:
First time committer here, thanks for clear instructions. Know where to look next time.

toredash committed:
docs: Update AWS IAM policy for cilium-operator

Update documentation related to configuration on AWS. Certain
IAM Policies must be in place if --clustername and --eni-gc-tags
are not set.

Incomplete docs got introduced after e66ed7fee9513b7ac23b3eb8327679a8121dd6c2

Signed-off-by: Tore S. Loenoey <tore.lonoy@gmail.com>
